### ISSUE TITLE
The proxy sometimes sometimes gets stuck "on"

### ISSUE THREAD

**Issue by jsrobson10** [wakeuteu/socksproxy-firefox-addon#7]
The proxy will sometimes continue to work even when its disabled and the browser is restarted. The proxy will become stuck "on" (it will display as off or on), and the plug-in has to be disabled and the browser restarted for it to work properly again.

**Comment by matthijskooijman** [comment]
TL;DR: Disabling the extension with the proxy still enabled seems to mess up Firefox' internal state, workaround and possible fix suggested at the bottom.

I had the same issue, except that restarting the browser did not help: as soon as I enabled the extension, the proxy would be enabled.

Investigating this a bit using the [extension debug tools](https://extensionworkshop.com/documentation/develop/debugging/), I discovered that the `originalProxySettings` that the extension requests and save on startup already had a proxy enabled, so the extension would end up switching beween the extension's proxy settings, and the "original" settings, which were identical.

Here's what the `proxySettings` requested from the browser looked like [here](https://github.com/wakeuteu/socksproxy-firefox-addon/blob/0a8f98e1b23171ac43fa215101b6fa6a48c38d87/background/socksproxy.js#L59):

![image](https://user-images.githubusercontent.com/194491/158359184-ce858e4f-1a76-410a-a0d9-076663d44399.png)

Which is actually weird, since this is how my browser settings looked like (note that there are proxy settings filled in, but the radio button is set to "No proxy", which does not match the `proxytype: "manual"` above).

![image](https://user-images.githubusercontent.com/194491/158360121-a06e6986-b7d4-4cfb-acff-8e7ee45f738d.png)

So it looks a bit like Firefox is lying to the extension about the current settings. However, this alone does not explain what I'm seeing entirely, since as soon as I enable the extension, the proxy is enabled. And looking at the code, the extension does not change the settings until you click the button, so you would then expect that the proxy remains disabled until you enable the extension, and then can no longer be disabled, but that's not what I saw.

This makes me suspect that Firefox actually tries to be smart and remembers the proxy settings last configured by the extension or something like that, and as soon as you enable the extension, it also switches the proxy settings back to what the extension last configured.

To test this theory, and disabled the extension and modified the proxy settings manually to use 127.0.0.2 and set them back to "No proxy". If FF really would save the last settings by the extension, I would expect nothing to change, and if there was some other issue (where FF maybe say non-empty proxy server and then autoswitches from "no proxy" to "manual" or so, I would get the same result but with a different IP). But doing this actually changed two things: The IP changed, but the type also changed to "none" instead of "manual"
![image](https://user-images.githubusercontent.com/194491/158362723-25e615e7-5243-4831-b9b0-7d2478c1b7e7.png)

So maybe FF also clears away any saved extension settings when you manually change the proxy settings?

In any case, after manually changing the settings, extension seemed to work again as expected (in the firefox network settings window, I could see the settings switch between manual/127.0.0.1 (as configured in the extension) and no proxy/127.0.0.2 (as I manually configured).

So maybe this issue is triggered by disabling the extension when it still has the proxy enabled? If I do that, I can see that indeed the network settings window still shows the extension-configured 127.0.0.1, but the type has switched from "manual" to "No proxy":

![image](https://user-images.githubusercontent.com/194491/158364173-d546f916-8aac-47b6-9507-2a5d8201be08.png)

Subsequently enabling the extension again indeed shows that the 127.0.0.1 are kept, but the type switches to "manual" when the extension is enabled: 

![image](https://user-images.githubusercontent.com/194491/158364551-02f1812c-4633-46ce-a52f-68bb0aa8ee02.png)

So that indeed seems to be the cause of (at least my version of) this problem. I guess a fix for this would be for the extension to disable the proxy (restore the original settings) just before the extension itself is disabled (haven't tried this, though).

So, summarizing:

**Cause**: Disabling the extension while the proxy is enabled (icon is green) makes the proxy settings sticky (always on as long as the extension is enabled)
**Workaround**: With the extension disabled, manually change the proxy settings (i.e. set to manual, change or clear the proxy host and/or port, set to "no proxy" again, save).
**Suggested fix**: When disabling the extension, disable the proxy first (restore original settings)

**Comment by wakeuteu** [comment]
Hi,
Thanks for the extensive feedback on this. It looks like my v1.1.1 has improved this bad behavior.
When I restart browser with extension "on", I restart it and it is still "on" **but the default config is not overridden anymore so when I turn on & off the extension again it is fine.**
I will make the currentState (enabled/disabled) persistent in storage, that should do the trick.
Best regards,
Anthony
